File: problem_solving/duplicates.py
def containsDuplicate(nums: list[int]) -> bool:
    counter = {n: 0 for n in nums}
    for i in range(0, len(nums)):
        counter[nums[i]]+=1
    
    for k, v in counter.items():
        if v > 1:
            return True
            
    return False

def containsDuplicate(nums: list[int]) -> bool:
    return len(nums) != len(set(nums))


print(containsDuplicate([1,2,2,3,4]))

File: problem_solving/duplicates.cpp
#include <iostream>
#include <vector>
#include <map>


using namespace std;


bool containsDuplicate(vector<int>& nums) {
    map<int, int> mp;
    for (int i: nums){
        mp[i]++;
    }
    for (auto i: mp){
        if (i.second > 1){
            return true;
        }
    }
    return false;
}

int main(){
    vector<int> in = {3,9,1,5,6};
    cout << containsDuplicate(in) << endl;
    
    return 0;
}